starknet_py/tests/e2e/client/full_node_test.py

```diff
@@ -23,51 +25,25 @@ async def test_node_get_transaction_by_block_hash_and_index(
         constructor_calldata=[],
         max_fee=0,
         signature=[],
-        # class_hash=class_hash,
+        class_hash=class_hash,
     )
 
 
 @pytest.mark.asyncio
-async def test_node_get_deploy_transaction_by_block_number_and_index(
-    deploy_transaction_hash, contract_address, rpc_client
+async def test_node_get_deploy_transaction_by_block_id_and_index(
+    deploy_transaction_hash, contract_address, rpc_client, class_hash
 ):
-    tx = await rpc_client.get_transaction_by_block_number(block_number=1, index=0)
+    tx = await rpc_client.get_transaction_by_block_id(block_number=1, index=0)
 
     assert tx == DeployTransaction(
         hash=deploy_transaction_hash,
         contract_address=contract_address,
         constructor_calldata=[],
         max_fee=0,
         signature=[],
-        # class_hash=class_hash,
-    )
-
-
-@pytest.mark.asyncio
-async def test_get_block_throws_on_no_block_hash_and_no_number(rpc_client):
-    with pytest.raises(ValueError) as exinfo:
-        await rpc_client.get_block()
-
-    assert "Either block_hash or block_number is required" in str(exinfo.value)
```

Comment:
Why was this test removed?

Reply:
Because get_block will not throw. We have default block_id="pending"